### PR TITLE
(Feat/month) Implement basic routing from month and month navigation via flux 

### PR DESCRIPTION
**Pull request by joshuaNewman10**

### CODE REVIEW

File: client/js/components/month/month.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-var MonthHeader = require('./month-header.js');
-var MonthData = require('../../monthData.js');
+var MonthStore = require('../../stores/month-store');
+var MonthHeader = require('./month-header');
+var MonthData = require('../../monthData');
 
 function monthData () {
   return {
-    gigs: MonthStore.getMonthData()
+    events: MonthStore.getMonthData()
```
Comment (kyleaowen) [inline client/js/components/month/month.js]: Having this 'events' property does not match up with the structure of the initial month data. Looking at the MonthEventsContainer component, you expect to loop through the monthWeeks, which will work fine with the initial data Month data the month component receives, but not work with the month data retrieved _onChange. 

``` javascript
{
  events: {
                week1: ...
                week2: ...
               ....
             }
}
```

vs 

``` javascript
{
   week1: ...
   week2: ...
    .... 
}
```

File: client/js/components/month/month-week-board.js [hunk 2]
```diff
@@ -1,15 +1,20 @@
 /** @jsx React.DOM */
 var React = require('react');
-var MonthDayCard = require('./month-day-card.js');
+var MonthDayCard = require('./month-day-card');
 var _ = require('underscore');
+var Link = require('react-router-component').Link
 
 var MonthWeekBoard = React.createClass({
   render: function () {
     var weekEvents = _.map(this.props.weekEvents, function(day, key) {
```
Comment (kyleaowen) [inline client/js/components/month/month-week-board.js]: the variable naming here threw me off a bit - you are mapping weekEvents, so I'd imagine that each item in weekEvents is an event. checking the data, it looks like weekEvents is an array which contains data for each Day (which is well labeled as 'day' in the params). To prevent this confusion later on, I think that we should change the variable 'weekEvents' to something more like 'weekData' which contains 'dayData'. thoughts? 

Also, for the finalized data, I think it'd be clearer for the weekData to be an object with properties for each day with values equal to that days data. Having it in an array just seems a little less robust (we may need to count on the order of the array, etc...)

File: client/js/stores/month-store.js [hunk 3]
```diff
@@ -2,7 +2,9 @@
 // var Dispatcher = require('../dispatcher/dispatcher');
```
Comment (kyleaowen) [inline client/js/stores/month-store.js:2]: we can remove this comment

File: client/js/components/month/month.js [hunk 4]
```diff
@@ -1,15 +1,9 @@
...
-var MonthData = require('../../monthData.js');
-
-function monthData () {
-  return {
-    gigs: MonthStore.getMonthData()
-  };
-}
+var MonthStore = require('../../stores/month-store');
+var MonthHeader = require('./month-header');
+var MonthData = require('../../monthData');
```
Comment (kyleaowen) [inline client/js/components/month/month.js:6]: the following suggested change may merit it's own PR to get flux initialization in place: 

monthData shouldn't be required into the month component. instead, trigger a ViewAction to get the data on componentWillMount or componentDidMount after adding the listener for the store. getInitialState should return an object with the default properties for this.state (here it could just be an empty object). In each subcomponent, we should use getDefaultProps to set defaults.

File: client/js/components/month/month-week-board.js (hunk 2 above)
Comment (joshuaNewman10) [inline client/js/components/month/month-week-board.js]: Agreed. I think the final naming convention will be largely informed by what the data is actually going to look like. For now im definitely happy to rename it to whatever is clearer for code reviews. And similar response to your second point, Id like to hold off on modifying the mock data until we have finalized version one of our schema and spoken to the company more about the data we can expect

File: client/js/components/month/month.js (hunk 4 above)
Comment (joshuaNewman10) [inline client/js/components/month/month.js:6]: Could catch, Ill submit another PR for the initilization now that flux has been implemented for the month componenet

**Comment by kyleaowen** [comment]
looks good!
